cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/MetadataDocument.java

```diff
@@ -142,11 +142,20 @@ public String toString() {
     private final String scope;
     private final String name;
     private final String value;
+    // the value field's numeric representation, if applicable
+    private final Long numericValue;
 
     Property(String scope, String name, String value) {
       this.scope = scope;
       this.name = name;
       this.value = value;
+      Long temp;
+      try {
+        temp = Long.parseLong(value);
```

Comment:
do you need a temp? Can't you just set `this.numericValue = Long.parseLong(value)`? And ignore the exception?

Reply:
I've tried a few things with initializing this field:
If I try `this.numericValue = Long.parseLong(value)` without a try-catch statement, CDAP throws many NumberFormatExceptions, and I can no longer add or delete properties.
If I try the code within a `try` block and do nothing within the `catch` block, the variable is not guaranteed to be initialized.
Then, if I try to assign it a value in either the `try` or `catch` block, the fact that the field is `final` causes an issue of possible double initialization.
For that reason, a temporary variable was the best solution I could come up with. Perhaps I should include a comment explaining why it's there?

Reply:
I address this in the latest commit by placing the try-catch logic in a different method, and assigning a value to numericValue based on the return of that method.